File: bin/pbr.py
#!/usr/bin/env python
#
# simple command to send note via PushBullet (http://pushbullet.com).
# Requires the installation of the pushbullet.py module from
# https://github.com/randomchars/pushbullet.py
#
# Author: Marc Dougherty <muncus@gmail.com>

import os
import pushbullet
import yaml

from optparse import OptionParser

parser = OptionParser()
parser.add_option("-t", "--title", dest="title", help="title for note")
parser.add_option("-b", "--body", dest="body", help="body for note")

(options, args) = parser.parse_args()

CONFIG_FILE = os.path.expanduser('~/.pushbullet')

config = {}

def get_api_key():
  if config.has_key('api_key'):
    return config["api_key"]
  return None


if os.path.exists(CONFIG_FILE):
  config = yaml.load(open(os.path.expanduser('~/.pushbullet')))


pb = pushbullet.PushBullet(get_api_key())
#TODO: expand this to be a particular device, or all, controlled by flags.
device = pb.devices[0]
device.push_note(options.title, options.body)

File: bin/pbr.rb
#!/usr/bin/ruby
# PushBullet Relay (aka - PBR)
# Push notification to your device using pushbullet's api.
# relies on the washbullet gem. 
# (gem install --user-install washbullet)

require 'washbullet'
require 'optparse'
require 'safe_yaml'
SafeYAML::OPTIONS[:default_mode] = :safe

configfile = File.expand_path("~/.pushbullet")

options = Hash.new('')

OptionParser.new do |opts|
  opts.on("-f", "--file FILE", "Config file containing API key") do |f|
    configfile = File.expand_path(f)
  end

  opts.on("-t", "--title TITLE", "Title for the push.") do |t|
    options[:title] = t
  end

  opts.on("-b", "--body BODY", "Push body.") do |b|
    options[:body] = b
  end

  opts.on("-l", "--link URL", "URL to push") do |b|
    options[:link] = b
  end
end.parse!

config = YAML.load_file(configfile)

#TODO:
# * device picker/ all devices
# * pull push content from flags.

c = Washbullet::Client.new(config['api_key'])
deviceid = c.devices[0]
if options[:link] != ''
  resp = c.push_link(
      receiver: deviceid,
      params: {
        title: options[:title],
        url: options[:link],
        body: options[:body]})
else
  resp = c.push_note(
      receiver: deviceid,
      params: {
        title: options[:title],
        body: options[:body]})
end

p resp
